src/hooks.h

```diff
@@ -0,0 +1,255 @@
+#pragma once
+
+#include "macro_functions.h"
+#include "plugin.h"
+#include "hook_signature_check.h"
+
+// A note to maintainers: How to add additional hooks
+//
+// 1) Add an appropriate non-virtual (important!) method
+//       to class KaleidoscopePlugin
+// 2) Add a virtual (important!) method with the same name and call signature
+//       to the abstract base class PluginHookAdapter__
+// 3) Add a hook specification to template class PluginHookAdapter
+//       via the HOOK_TASK macro
+// 4) Add an implemenation of the hook to template class PluginHookAdapter
+//       that references the name given in HOOK_TASK
+
```

Comment:
I can't figure out what to do for the last step here. Or maybe I got that one, but not step 3? Can you point to specific lines in specific files for these two steps for `eventHandlerHook`?

Reply:
I think I've figured it out now; step 4 is the line following step 3. I think it would be less confusing to combine these into a single step, because it's just one place in the code.